metadata/metadata_test.go

```diff
@@ -207,4 +295,203 @@ var _ = Describe("Metadata", func() {
 			}))
 		})
 	})
+
+	When("CopyAnnotationsWithPrefixReplacement/CopyAnnotationsByPrefix is called", func() {
+		It("should return all the annotations matching the given prefix", func() {
+			podSource := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{"foo": "bar", "baz": "qux"},
+				},
+			}
+			podDest := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{"quux": "corge"},
+				},
+			}
+			CopyAnnotationsWithPrefixReplacement(podSource, podDest, "fo", "ba")
+			CopyAnnotationsByPrefix(podSource, podDest, "fo")
+			Expect(podDest.Annotations).To(gstruct.MatchAllKeys(gstruct.Keys{
+				"foo":  Equal("bar"),
+				"bao":  Equal("bar"),
+				"quux": Equal("corge"),
+			}))
+		})
+
+		It("should copy all the annotations matching the given prefix when Annotations in Dest is empty", func() {
+			podSource := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{"foo": "bar", "baz": "qux"},
+				},
+			}
+			podDest := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{},
+			}
+			err := CopyAnnotationsWithPrefixReplacement(podSource, podDest, "fo", "ba")
```

Comment:
`Expect(CopyAnnotationsWithPrefixReplacement(podSource, podDest, "fo", "ba")).To(Succeess())` and remove the err check.

Reply:
Updated, thanks!